reference/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+pub mod sparse_merkle_tree;
+
+use std::marker::PhantomData;
+
+use light_bounded_vec::{BoundedVec, BoundedVecError};
+use light_hasher::{errors::HasherError, Hasher};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ReferenceMerkleTreeError {
+    #[error("Leaf {0} does not exist")]
+    LeafDoesNotExist(usize),
+    #[error("Hasher error: {0}")]
+    Hasher(#[from] HasherError),
+    #[error("Invalid proof length provided: {0} required {1}")]
+    InvalidProofLength(usize, usize),
+}
+
+#[derive(Debug, Clone)]
```

Comment:
I would like to keep the reference tree in the mono repo because we have been modifying it a lot for testing lately